code/entities/chair/Armrest.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Sandbox;
+
+namespace Cinema;
+
+public partial class Armrest : BaseNetworkable
+{
+    public enum States : int
+    {
+        Lowered,
+        Raising,
+        Raised,
+        Lowering
+    }
+
+    public enum Sides : int
+    {
+        Left,
+        Right
+    }
+
+    [Net]
+    public CinemaChair Chair { get; set; }
+    [Net]
+    public ModelEntity Cupholded { get; set; }
+    [Net]
+    public States State { get; set; } = States.Lowered;
+    [Net]
+    public Sides Side { get; set; }
+
+    public string CupholderBoneName => Side switch
+    {
+        Sides.Left  => "cupholder_l",
+        _           => "cupholder_r"
+    };
+
+    public string ArmrestToggleParameter => Side switch
+    {
+        Sides.Left => "toggle_left_armrest",
+        _ => "toggle_right_armrest"
+    };
+
+    public void HandleAnimTag(string tag)
+    {
+        if (tag.Contains("launch"))
+        {
+            Launch();
+            State = States.Raised;
+        }
+        else if (tag.Contains("lowered"))
+        {
+            State = States.Lowered;
+        }
+    }
+
+    public void Toggle()
+    {
+        var originalValue = Chair.GetAnimParameterBool(ArmrestToggleParameter);
+        Chair.SetAnimParameter(ArmrestToggleParameter, !originalValue);
+        // If toggle_seat was true but now is false, we are now lowering the seat.
+        State = originalValue
+            ? States.Lowering
+            : States.Raising;
+    }
+
+    public void CupholdEntity(ModelEntity entity)
+    {
+        if (State != States.Lowered)
+        {
+            return;
```

Comment:
Should this maybe return `false` to say "no we did not put the entity in the cupholder"?

Reply:
I went ahead and changed this method to `TryHoldEntity`, which returns whether the `entity` argument actually gets held.